src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/internal/core/java/corrections/JavaDiagnosticsDefinition.java

```diff
@@ -50,49 +51,48 @@ public final class JavaDiagnosticsDefinition extends BaseKeyedLazyInstance<IJava
-        try {
-            return getInstance().isAdaptedForDiagnostics(context);
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling isAdaptedForDiagnostics", e);
-            return false;
-        }
+        return ExceptionUtil.executeWithExceptionHandling(
+                () -> getInstance().isAdaptedForDiagnostics(context),
+                () -> false,  // Fallback value in case of exception
+                e -> LOGGER.log(Level.WARNING, "Error while calling isAdaptedForDiagnostics", e)
+        );
     }
 
     @Override
     public void beginDiagnostics(JavaDiagnosticsContext context) {
-        try {
-            getInstance().beginDiagnostics(context);
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling beginDiagnostics", e);
-        }
+        ExceptionUtil.executeWithExceptionHandling(
+            () -> {
+                getInstance().beginDiagnostics(context);
+                return true;
+            },
+                null,
+            e -> LOGGER.log(Level.WARNING, "Error while calling beginDiagnostics", e)
+        );
     }
 
     @Override
     public List<Diagnostic> collectDiagnostics(JavaDiagnosticsContext context) {
-        try {
-            List<Diagnostic> diagnostics = getInstance().collectDiagnostics(context);
-            return diagnostics != null ? diagnostics : Collections.emptyList();
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling collectDiagnostics", e);
-            return Collections.emptyList();
-        }
+        return ExceptionUtil.executeWithExceptionHandling(
+                () -> {
+                    List<Diagnostic> diagnostics = getInstance().collectDiagnostics(context);
+                    return diagnostics != null ? diagnostics : Collections.emptyList();
+                },
+                Collections::emptyList,
+                e -> LOGGER.log(Level.WARNING, "Error while calling collectDiagnostics", e)
+        );
     }
 
+
     @Override
     public void endDiagnostics(JavaDiagnosticsContext context) {
-        try {
-            getInstance().endDiagnostics(context);
-        } catch (IndexNotReadyException | ProcessCanceledException | CancellationException e) {
-            throw e;
-        } catch (Exception e) {
-            LOGGER.log(Level.WARNING, "Error while calling endDiagnostics", e);
-        }
+        ExceptionUtil.executeWithExceptionHandling(
+            () -> {
+                getInstance().endDiagnostics(context);
+                return true;
+            },
+                null,
```

Comment:
`NullPointerException`? See my previous comment.

Reply:
Did change similar to above comment

```
@Override
    public void endDiagnostics(JavaDiagnosticsContext context) {
        ExceptionUtil.executeWithExceptionHandling(
            () -> {
                getInstance().endDiagnostics(context);
                return true;
            },
            e -> {
                LOGGER.log(Level.WARNING, "Error while calling endDiagnostics", e);
                return null;
            }
        );
    }
```